### DISCUSSION TITLE
Problem: Data in change.Payload.Data.Record is null when using AddPostgresChangeHandler in C#

### DISCUSSION THREAD

**Discussion by zhucode** [supabase/supabase#28085] [Questions]
I am using the Supabase Realtime library in a C# project to react to changes in a Postgres database. When registering a handler for changes with AddPostgresChangeHandler, I find that the data in change.Payload.Data.Record is always null.

Here is a code example showing how I register the handler:

```
var channel = supabase.Realtime.Channel("realtime", "public", "users", "email=u.assmus@gmx.net");
channel.AddPostgresChangeHandler(ListenType.All, (sender, change) =>
{
    // The changed record 
    Debug.WriteLine(change.Payload.Data.Record);
});
await channel.Subscribe();
```

The issue is that change.Payload.Data.Record is always null. To investigate the problem, I used a method to access the internal JSON data:

```
channel.AddPostgresChangeHandler(ListenType.All, (sender, change) =>
{
    var jsonProperty = change.GetType().GetProperty("Json", BindingFlags.NonPublic | BindingFlags.Instance);
    string jsonString = jsonProperty?.GetValue(change)?.ToString();

    var json = JObject.Parse(jsonString);
    jsonString = json.ToString(Newtonsoft.Json.Formatting.Indented);

    var payloadData = json["payload"]?["data"];
    if (payloadData != null)
    {
        var updatedUser = payloadData["record"];
        Console.WriteLine("Updated User:");
        Console.WriteLine($"ID: {updatedUser["id"]}");
        Console.WriteLine($"Name: {updatedUser["name"]}");
        Console.WriteLine($"Email: {updatedUser["email"]}");
        Console.WriteLine($"StripeId: {updatedUser["stripeCustomerId"]}");
        Console.WriteLine($"UUID: {updatedUser["UUID"]}");
    }
});
```

Using this method, I see that the data is indeed present, but the JSON data is not being correctly mapped to the Payload.Data class. It appears to be an issue with deserializing the JSON data into the Payload.Data class.

Expected Behavior:
change.Payload.Data.Record should contain the changed records as shown in the JSON data.

Actual Behavior:
change.Payload.Data.Record is always null.

Does anyone know how to resolve this issue?
Thank you!

**Comment by GaryAustin1** [comment]
You might want to file this (and search issues) in the C# library as it is a community project.   https://github.com/supabase-community/realtime-csharp

**Comment by muttkiddo** [comment]
@zhucode your code saved my day, also was getting null from the Data.Record all the time, your solution worked around perfectly, I hope @GaryAustin1 supabase can fix this!!!

**Reply by GaryAustin1** [reply]
That is a community library and maintained directly by Supabase.  Also I'm not an employee in case you were implying I could do something about it.  You two might post in that repository or generate a PR to fix it.
